framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/IdentityModelAuthenticationService.cs

```diff
@@ -106,40 +110,39 @@ private IdentityClientConfiguration GetClientConfiguration(string identityClient
         protected virtual async Task<DiscoveryDocumentResponse> GetDiscoveryResponse(
             IdentityClientConfiguration configuration)
         {
-            using (var httpClient = HttpClientFactory.CreateClient())
+            var httpClient = HttpClientFactory.CreateClient(HttpClientName);
+            return await httpClient.GetDiscoveryDocumentAsync(new DiscoveryDocumentRequest
             {
-                return await httpClient.GetDiscoveryDocumentAsync(new DiscoveryDocumentRequest
+                Address = configuration.Authority,
+                Policy =
                 {
-                    Address = configuration.Authority,
-                    Policy =
-                    {
-                        RequireHttps = configuration.RequireHttps
-                    }
-                });
-            }
+                    RequireHttps = configuration.RequireHttps
+                }
+            });
         }
 
         protected virtual async Task<TokenResponse> GetTokenResponse(
             DiscoveryDocumentResponse discoveryResponse,
             IdentityClientConfiguration configuration)
         {
-            using (var httpClient = HttpClientFactory.CreateClient())
+            var httpClient = HttpClientFactory.CreateClient(HttpClientName);
```

Comment:
Why not disposing the client (I didn't know if it is needed when you use HttpClientFactory.CreateClient)?
If you are re-using the same instance, should we care about multi-threading. In addition, should we check if tenantId added before and override it?
Can you explain a bit what this PR is trying to do.

Reply:
There is no need to dispose of the HttpClient instances from HttpClientFactory. Disposal will not actually do anything in this case because the factory manages the handler and connection lifetimes and not the HttpClient instances. Calling dispose on HttpClient has no effect for instances provided by the factory and is simply redundant code.

Related https://support.abp.io/QA/Questions/171/How-to-call-abp-vnext-dynamic--api#answer-31b96b0e-ecd5-13ea-abca-39f56671b24d